Permanent/ViewControllers/TagDetailsViewController.swift

```diff
@@ -0,0 +1,234 @@
+            self.sortedArray = tagsArchive.map( { SortedTagVO(tagVO: $0) } )
+            
+            if let tags = self.tagVOS {
+                for tagItem in tags {
+                    var idx : Int?
+                    idx = self.sortedArray.firstIndex(where: { $0.tagVO.tagVO.name == tagItem.name})
+
+                    if idx != nil { self.sortedArray[idx!].checked = true }
+                }
+            }
+            self.sortedArray.sort { (_, arg0) -> Bool in !arg0.checked }
+            self.filteredTagVO = self.sortedArray.map({ (item) -> TagVO in
+                return item.tagVO
+            })
+
+            self.tagsCollectionView.reloadData()
+            self.hideSpinner()
+        })
+    }
+}
+
+extension TagDetailsViewController: UICollectionViewDelegate, UICollectionViewDataSource, UICollectionViewDelegateFlowLayout {
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        return filteredTagVO.count
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        let cell = collectionView.dequeueReusableCell(withReuseIdentifier: TagCollectionViewCell.identifier, for: indexPath) as! TagCollectionViewCell
+        
+        if let tagName = filteredTagVO[indexPath.row].tagVO.name {
+            cell.configure(name: tagName, isVisible: sortedArray[indexPath.row].checked)
+        }
+        return cell
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
+        let cell = collectionView.cellForItem(at: indexPath) as! TagCollectionViewCell
+        
+        UIView.animate(withDuration: 0.05, animations: { cell.alpha = 0.5 }) { (completed) in
+            UIView.animate(withDuration: 0.05, animations: { cell.alpha = 1 })  { (completed) in
+                collectionView.reloadData()
+            }
+        }
+        
+        sortedArray[indexPath.row].checked = !sortedArray[indexPath.row].checked
+        sortedArray[indexPath.row].forAdding = sortedArray[indexPath.row].checked
+        sortedArray[indexPath.row].forRemoval = !sortedArray[indexPath.row].checked
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, minimumInteritemSpacingForSectionAt section: Int) -> CGFloat {
+        return 10
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, minimumLineSpacingForSectionAt section: Int) -> CGFloat {
+        return 10
+    }
+   
+    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, sizeForItemAt indexPath: IndexPath) -> CGSize {
+        let additionalSpace: CGFloat = (sortedArray[indexPath.row].checked == true) ? ( 45 ) : ( 35 )
+        
+        if  let name = filteredTagVO[indexPath.row].tagVO.name {
```

Comment:
if  let name = filteredTagVO[indexPath.row].tagVO.name,
            let isChecked = sortedArray.first(where: { $0.tagVO.tagVO.name == name })?.checked {
            let additionalSpace: CGFloat = isChecked ? ( 45 ) : ( 35 )

Reply:
Adapted logic to use isChecked variable.